src/EditorFeatures/Core/Implementation/Formatting/Indentation/AbstractSmartTokenFormatterCommandHandler.cs

```diff
@@ -1,353 +0,0 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Threading;
-using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.Formatting;
-using Microsoft.CodeAnalysis.Formatting.Rules;
-using Microsoft.CodeAnalysis.Options;
-using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.Text;
-using Microsoft.CodeAnalysis.Text.Shared.Extensions;
-using Microsoft.VisualStudio.Commanding;
-using Microsoft.VisualStudio.Text;
-using Microsoft.VisualStudio.Text.Editor;
-using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
-using Microsoft.VisualStudio.Text.Editor.OptionsExtensionMethods;
-using Microsoft.VisualStudio.Text.Operations;
-using Roslyn.Utilities;
-using VSCommanding = Microsoft.VisualStudio.Commanding;
-
-namespace Microsoft.CodeAnalysis.Editor.Implementation.Formatting.Indentation
-{
-    internal abstract class AbstractSmartTokenFormatterCommandHandler :
-        IChainedCommandHandler<ReturnKeyCommandArgs>
```

Comment:
completely uneeded.  This just replicates the functionality that VS and the ISmartIndentProvider already take care of for us.

Reply:
are you sure? I remember you and sam arguing that formatting and smart identation is so important and related to musle memory so unless we can prove it doesn't change behavior, we should try to change it. is that now changed?

Reply:
I can't prove a negative.  But i can provide a lot of evidence this is safe.  Specifically, the other class only handled hitting enter after typing `using (...)`.  We've already discussed and designed a new approach for handling that that hte IDE is fine with.  Also, we have many tests covering indentation behavior, and we added more tests during the work to remov ethe need for `AbstractSmartTokenFormatterCommandHandler`.  These help validate that the behavior we have is the one we want.